pkg/agent/knapsack/knapsack.go

```diff
@@ -26,23 +30,45 @@ type knapsack struct {
 	// remove this field and prevent "leaking" bbolt into places it doesn't need to.
 	db *bbolt.DB
 
+	slogger       *multislogger.MultiSlogger
+	systemSlogger *slog.Logger
+
 	// This struct is a work in progress, and will be iteratively added to as needs arise.
 	// Some potential future additions include:
 	// Querier
 }
 
-func New(stores map[storage.Store]types.KVStore, flags types.Flags, db *bbolt.DB) *knapsack {
+func New(stores map[storage.Store]types.KVStore, flags types.Flags, db *bbolt.DB, slogger *multislogger.MultiSlogger) *knapsack {
 	k := &knapsack{
-		db:     db,
-		flags:  flags,
-		stores: stores,
+		db:      db,
+		flags:   flags,
+		stores:  stores,
+		slogger: slogger,
 	}
 
+	// TODO: figure out windows thang
+	k.systemSlogger = slog.New(slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{
```

Comment:
Not sure if it'll make sense to pass the system slogger in at creation time, or create it here and add to it

Reply:
Updated to create in main and pass to run launcher